sideline.el

```diff
@@ -236,15 +236,21 @@
   `(when (buffer-live-p ,buffer-or-name)
      (with-current-buffer ,buffer-or-name ,@body)))
 
-;; TODO: Use function `string-pixel-width' after 29.1
-(defun sideline--string-pixel-width (str)
-  "Return the width of STR in pixels."
-  ;; Text properties may effect the length, remove it!
-  (let ((str (substring-no-properties str)))
-    (if (fboundp #'string-pixel-width)
-        (string-pixel-width str)
-      (require 'shr)
-      (shr-string-pixel-width str))))
+(defun sideline--string-pixel-width (string)
+  "Return the width of STRING in pixels.
+
+Sideline's own version of `string-pixel-width'
+which respect the `face-remapping-alist'."
+  (if (zerop (length string))
+      0
+    (let ((remapping-alist face-remapping-alist))
+      ;; Prevent use original buffer name for minimal side-effects
+      (with-current-buffer (get-buffer-create " *sideline-string-pixel-width*")
```

Comment:
Can we use `with-temp-buffer` here? 🤔

Reply:
Ref to the built-in `string-pixel-width` function:
```elisp
(defun string-pixel-width (string)
  "Return the width of STRING in pixels."
  (if (zerop (length string))
      0
    ;; Keeping a work buffer around is more efficient than creating a
    ;; new temporary buffer.
    (with-current-buffer (get-buffer-create " *string-pixel-width*")
      ;; If `display-line-numbers' is enabled in internal buffers
      ;; (e.g. globally), it breaks width calculation (bug#59311)
      (setq-local display-line-numbers nil)
      (delete-region (point-min) (point-max))
      ;; Disable line-prefix and wrap-prefix, for the same reason.
      (setq line-prefix nil
	    wrap-prefix nil)
      (insert (propertize string 'line-prefix nil 'wrap-prefix nil))
      (car (buffer-text-pixel-size nil nil t)))))
```